src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationMiddleware.cs

```diff
@@ -42,6 +43,7 @@ public class OpenIdConnectAuthenticationMiddleware : AuthenticationMiddleware<Op
             [NotNull] IDataProtectionProvider dataProtectionProvider,
             [NotNull] ILoggerFactory loggerFactory,
             [NotNull] IUrlEncoder encoder,
+            [NotNull] IDistributedCache cache,
```

Comment:
@Tratcher I used this approach for `HtmlEncoder` in `OpenIdConnectServerMiddleware`, but TBH, I was not totally satisfied... it rather sounded like an anti-pattern: https://github.com/aspnet-contrib/AspNet.Security.OpenIdConnect.Server/blob/vNext/src/AspNet.Security.OpenIdConnect.Server/OpenIdConnectServerMiddleware.cs#L65